src/main/kotlin/org/virtuslab/pulumikotlin/codegen/step3codegen/resources/ResourceGenerator.kt

```diff
@@ -139,20 +146,22 @@ object ResourceGenerator {
             .addDocs("@param block The arguments to use to populate this resource's properties.")
             .build()
 
-        requireNotNull(::opts) // to ensure compile-time safety
-        val optsCreationFunction = MemberName("com.pulumi.kotlin.options", "opts")
+        val optsClass = determineOptsClass(superclass)
+        val optsBuilderClass = determineOptsBuilderClass(superclass)
+        
+        val optsFunctionName = MemberName(optsClass, "opts")
 
         val optsFunction = FunSpec
             .builder("opts")
             .addModifiers(KModifier.SUSPEND)
             .addParameter(
                 "block",
                 LambdaTypeName.get(
-                    customResourceOptionsBuilderClass(),
+                    optsBuilderClass,
                     returnType = UNIT,
                 ).copy(suspending = true),
             )
-            .addStatement("this.opts = %M(block)", optsCreationFunction)
+            .addStatement("%M(block)", optsFunctionName)
```

Comment:
I didn't dig much, so might be a dumb question: why is `this.opts = ` not needed anymore?

Reply:
It is haha, fixed it. At first I wanted to remove it and change the method body to something like this, same as with `args`:
```kotlin
val builder = CustomResourceOptionsBuilder()
block(builder)
this.opts = builder.build()
```
but that required exposing the `build` function from the common SDK, so I decided to keep it internal and instead create these two static methods